src/device/GPU_Device.cc

```diff
@@ -218,152 +151,141 @@ void GPU_Device::checkForCudaError(cudaError_enum const errorCode) {
     break;
-  case CUDA_ERROR_UNKNOWN:
+  case cudaErrorUnknown:
     message = std::string("Unknown error. ");
     break;
   default:
-    // CUDA_ERROR_PROFILER_DISABLED
-    // CUDA_ERROR_PROFILER_NOT_INITIALIZED
-    // CUDA_ERROR_PROFILER_ALREADY_STARTED
-    // CUDA_ERROR_PROFILER_ALREADY_STOPPED
-    // CUDA_ERROR_NOT_MAPPED_AS_ARRAY
-    // CUDA_ERROR_NOT_MAPPED_AS_POINTER
-    // CUDA_ERROR_ECC_UNCORRECTABLE
-    // CUDA_ERROR_UNSUPPORTED_LIMIT
-    // CUDA_ERROR_CONTEXT_ALREADY_IN_USE
-    // CUDA_ERROR_SHARED_OBJECT_SYMBOL_NOT_FOUND
-    // CUDA_ERROR_SHARED_OBJECT_INIT_FAILED
-    // CUDA_ERROR_OPERATING_SYSTEM
-    // CUDA_ERROR_PEER_ACCESS_ALREADY_ENABLED
-    // CUDA_ERROR_PEER_ACCESS_NOT_ENABLED
-    // CUDA_ERROR_PEER_MEMORY_ALREADY_REGISTERED
-    // CUDA_ERROR_PEER_MEMORY_NOT_REGISTERED
-    // CUDA_ERROR_PRIMARY_CONTEXT_ACTIVE
-    // CUDA_ERROR_CONTEXT_IS_DESTROYED
+    // cudaErrorProfilerDisabled
+    // cudaErrorNotMappedAsArray
+    // cudaErrorNotMappedAsPointer
+    // cudaErrorECCUncorrectable
+    // cudaErrorUnsupportedLimit
+    // cudaErrorSharedObjectSymbolNotFound
+    // cudaErrorSharedObjectInitFailed
+    // cudaErrorOperatingSystem
+    // cudaErrorPeerAccessAlreadyEnabled
+    // cudaErrorPeerAccessUnsupported
+    // cudaErrorPeerAccessNotEnabled
+    // cudaErrorContextIsDestroyed
     message = std::string("Unknown error. ");
     break;
   }
+  */
   return message;
 }
 
 //---------------------------------------------------------------------------//
 /*!
- * \brief Wrap the cuMemAlloc funtion to include error checking
+ * \brief Wrap the cudaMemAlloc funtion to include error checking
  *
  * \param nbytes number of bytes to allocate (e.g.: len*sizeof(double) ).
  * \return GPU device pointer to allocated memory.
  */
 
-CUdeviceptr GPU_Device::MemAlloc(unsigned const nbytes) {
-  CUdeviceptr ptr;
-  cudaError_enum err = cuMemAlloc(&ptr, nbytes);
+void *GPU_Device::MemAlloc(unsigned const nbytes) {
+  void *ptr;
+  cudaError_t err = cudaMalloc(&ptr, nbytes);
   checkForCudaError(err);
   return ptr;
 }
 
-void GPU_Device::MemcpyHtoD(CUdeviceptr ptr, void const *loc, unsigned nbytes) {
-  cudaError_enum err = cuMemcpyHtoD(ptr, loc, nbytes);
+void GPU_Device::MemcpyHtoD(void *ptr, void const *loc, unsigned nbytes) {
+  cudaError_t err = cudaMemcpy(ptr, loc, nbytes, cudaMemcpyHostToDevice);
   checkForCudaError(err);
   return;
 }
 
-void GPU_Device::MemcpyDtoH(void *loc, CUdeviceptr ptr, unsigned nbytes) {
-  cudaError_enum err = cuMemcpyDtoH(loc, ptr, nbytes);
+void GPU_Device::MemcpyDtoH(void *loc, void *ptr, unsigned nbytes) {
+  cudaError_t err = cudaMemcpy(loc, ptr, nbytes, cudaMemcpyDeviceToHost);
   checkForCudaError(err);
   return;
 }
 
-void GPU_Device::MemFree(CUdeviceptr ptr) {
-  cudaError_enum err = cuMemFree(ptr);
+void GPU_Device::MemFree(void *ptr) {
+  cudaError_t err = cudaFree(ptr);
   checkForCudaError(err);
   return;
 }
```

Comment:
Great work reconstructing this file!  I don't like the `void*` stuff, but if that is what is required, then we'll live with it.

Reply:
Yeah, the `CUdeviceptr` is part of the driver API. It's defined as:

"CUDA device pointer CUdeviceptr is defined as an unsigned integer type whose size matches the size of a pointer on the target platform."

I'm not sure how different that was from a raw pointer?